Zilliqa/contracts/compressed/minimize_proxy.scilla

```diff
@@ -157,33 +185,52 @@ transition MintCallBack(to: ByStr20, new_to_bal: Uint128, new_supply : Uint128)
 isCurrImpl _sender;
 
 balances[to] := new_to_bal;
-totalSupply := new_supply
+total_supply := new_supply
 end
 
 transition IncreaseAllowance (spender : ByStr20, amount : Uint128)
 current_impl <- implementation;
+
+option_allowance <- allowances[_sender][spender];
+allowance = deconstruct_option_uint128 option_allowance;
+
 msg = {_tag : "IncreaseAllowance"; _recipient : current_impl; _amount : zero;
-spender : spender; amount : amount; initiator : _sender};
+spender : spender; amount : amount; initiator : _sender; current_allowance : allowance};
 msgs = one_msg msg;
 send msgs
 end
 
 transition DecreaseAllowance (spender : ByStr20, amount : Uint128)
 current_impl <- implementation;
+
+option_allowance <- allowances[_sender][spender];
+allowance = deconstruct_option_uint128 option_allowance;
+
 msg = {_tag : "DecreaseAllowance"; _recipient : current_impl; _amount : zero;
-spender : spender; amount : amount; initiator : _sender};
+spender : spender; amount : amount; initiator : _sender; current_allowance : allowance};
 msgs = one_msg msg;
 send msgs
 end
 
+transition AllowanceCallBack(initiator : ByStr20, spender : ByStr20, new_allowance : Uint128)
+isCurrImpl _sender;
+
+allowances[initiator][spender] := new_allowance
```

Comment:
I'm thinking of changing this because `initiator` and `spender` doesn't work very well for transferFrom. 

In Increase/DecreaseAllowance, `initiator` is the owner and `spender` is the `spender` so when you look at the transaction receipt, it looks fine. 

In transferFrom, the one who initiates the transaction (i.e. _sender of the transaction to the proxy) is the spender so when you look at the txn receipt, `initiator` is from address, and `spender` is the original initiator so it gets a bit confusing because the from address didn't initiate the txn. 

Maybe renaming `initiator` to `owner` might make more sense. Thoughts? This requires redeploying our proxy token contract.

Reply:
Allowance is a ZRC2 thing right? Does ZRC2 standard not have any recommendation for this? 🤔

Reply:
Usually `allowances` is used for DEX approval, so the DEX can spend on the owner's behalf. For ZRC-2, we recognise the owner as the spender as you are still sending the subsequent txn to ask the DEX to invoke `TransferFrom`.

Reply:
Ah... so if owner asks the dex to do a transferFrom, owner is the "real" initiator. That makes sense! Even on ETH, I don't normally see people using transferFrom outside of dapps so perhaps, this is probably okay ba.